### PR TITLE
README Maintenance

### PR DESCRIPTION
**Pull request by ElectronicsArchiver**
Please check the **[Preview]**.

### Changes

- Converted raw links into referenced ones
- Added badges for website, matrix, libera, license
- General formatting / spacing
- Added syntax highlighting for code blocks

[Preview]: https://github.com/ElectronicsArchiver/fwupd

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
 - Libera IRC channel `#fwupd`.
-  You can join through Libera's [IRC](https://libera.chat/)
-  or via the IRC bridge on [Matrix](https://matrix.to/#/#fwupd:libera.chat).
+  You can join through Libera's [IRC]
+  or via the IRC bridge on [Matrix].
 
 ## Compiling
 
-The most up to date compilation instructions are available in the [Wiki](https://github.com/fwupd/fwupd/wiki/Compilation).
+The most up to date compilation instructions are available in the [Wiki][Compiling].
```
Comment (hughsie) [inline README.md:43]: I'm not really a fan of the detached links; it makes it a bit hard to follow when using it as a text file imho. @superm1 your opinion?

Comment (ElectronicsArchiver) [reply README.md:43]: On that topic, I have a **[little something]**.

[little something]: https://github.com/MarkedDown/Links

Comment (superm1) [reply README.md:43]: I personally open markdown in vscode with markdown preview all the time. So I am indifferent.

**Comment by superm1** [comment]
Would you mind prefixing all of your commits messages with "trivial:"? This will be helpful to not mark this in the changelog generation.

**Comment by ElectronicsArchiver** [comment]
I'd like to request a rerun of the workflow~
Hope the linter is finally satisfied ..

**Comment by ElectronicsArchiver** [comment]
Am I blind or does that error log not show what the linting problem is .. ?

**Comment by superm1** [comment]
> Am I blind or does that error log not show what the linting problem is .. ?
> 
> https://github.com/fwupd/fwupd/runs/6056754509?check_suite_focus=true

I suggest to run markdownlint locally to see what fails.

**Comment by superm1** [comment]
I checked out your branch and here is the stuff that the pre-commit rule flags.  It looks like whitespace problems to me.  Use my other comment and you can set up pre-commit locally to fix it really easily.

```
diff --git a/README.md b/README.md
index 8bc0e4c22..f4037f9b7 100644
--- a/README.md
+++ b/README.md
@@ -11,18 +11,18 @@
 
 <br>
 
-[![Badge Website]][Website] 
-[![Badge Matrix]][Matrix] 
+[![Badge Website]][Website]
+[![Badge Matrix]][Matrix]
 [![Badge IRC]][IRC]
 
 <br>
 
 ---
 
-[![Badge Build]][Build Status] 
-[![Badge CodeQL]][CodeQL] 
-[![Badge Coverity]][Coverity] 
-[![Badge Fuzzing]][Fuzzing] 
+[![Badge Build]][Build Status]
+[![Badge CodeQL]][CodeQL]
+[![Badge Coverity]][Coverity]
+[![Badge Fuzzing]][Fuzzing]
 [![Badge CircleCI]][CircleCI]
 
 ---
@@ -235,7 +235,6 @@ ninja install
 ninja fuzz-firmware
 ninja fuzz-tpm-eventlog
 ```
-    
 
 
 
```

**Comment by superm1** [comment]
### Thanks for your commit
The fwupd project has had a style change starting with fwupd 1.6.x.  The style is summarized in https://github.com/fwupd/fwupd/blob/main/CONTRIBUTING.md and it appears your current pull request is inconsistent with the new style.

**Clang Format**
Fortunately we have added a `.clang-format` and various helper scripts to help you fix your pull request.  Follow these steps to configure the environment and fix your code.
1. Run `./contrib/setup` to set up [pre-commit](https://pre-commit.com/) locally.  This will help ensure that you have all the build dependencies necessary and automatically reformat code in the future.
2. Run `./contrib/reformat-code.py main` to reformat all code that has changed in your local branch relative to `main`
3. Run `git commit add .; git commit --amend` or use `git rebase -i` to amend your commit(s).
 
**Pre-commit overview**
The pre-commit rules are available at https://github.com/fwupd/fwupd/blob/main/.pre-commit-config.yaml and will check all languages and text for style.  After you have run `./contrib/setup` any future commits will check the diff relative to all of these rules.

**Comment by ElectronicsArchiver** [comment]
And how do I **remove** that again?

File: README.md (hunk 1 above)
Comment (hughsie) [reply README.md:43]: To be honest, I'm not a fan of the detached links; I edit and view most of the markdown files in a plain text editor, and having to scroll down the file to find a link doesn't really work for me. I don't agree the indirection cleans up the file, especially when most of the links are `[link description][link anchor]` and the `link anchor` is often as long as the original URL was.

**Comment by superm1** [comment]
> And how do I **remove** that again?

Remove pre-commit?  It's installed into your checkout.  You shouldn't have to remove it.   As this is all whitespace problems then I think you can also solve it by setting your editor to strip trailing whitespace.

**Comment by ElectronicsArchiver** [comment]
No, I meant how I could easily remove the pre-commit stuff~

**Comment by superm1** [comment]
> No, I meant how I could easily remove the pre-commit stuff~

Pre-commit is mandatory for merging changes.

**Comment by ElectronicsArchiver** [comment]
Of course I meant locally~

Btw, **which** markdownlint are you using,
I installed the original ruby gem but I don't
find anything about using a `json` file for 
the styles config, only a `ruby` one. . .

**Comment by ElectronicsArchiver** [comment]
I tried out multiple now, I guess it's the nodejs one?
At least that one works with json files

**Comment by ElectronicsArchiver** [comment]
Oh sorry, forgot to add `trivial` to the commits name..

File: .gitignore [hunk 2]
```diff
@@ -36,3 +36,7 @@ plugins/uefi-dbx/tests/
 .buildconfig
 .ossfuzz
 *.rej
+
+node_modules
+package-lock.json
+package.json
```
Comment (superm1) [inline .gitignore:42]: Please end with a blank new line

Comment (ElectronicsArchiver) [reply .gitignore:42]: Sure?

Comment (superm1) [reply .gitignore]: It previously did before your change, it should continue to.
